bluepyopt/deapext/tools/selIBEA.py

```diff
@@ -68,52 +54,37 @@ def selIBEA(population, mu, alpha=None, kappa=.05, tournament_n=4):
     return parents
 
 
-def _calc_box_bounds(population_matrix):
-    """Calculate the minimal square bounding box of the objectives"""
-
-    # Calculate the min/max over the columns
-    min_bounds = list(numpy.min(population_matrix, axis=0))
-    max_bounds = list(numpy.max(population_matrix, axis=0))
-
-    # Return, parse to a list (indicators need lists, not numpy arrays)
-    return list(min_bounds), list(max_bounds)
-
-
-def _calc_fitness_components(
-        population_matrix,
-        min_box_bounds,
-        max_box_bounds,
-        kappa=None):
+def _calc_fitness_components(population, kappa):
     """returns an N * N numpy array of doubles, which is their IBEA fitness """
+    # DEAP selector are supposed to maximise the objective values
+    # We take the negative objectives because this algorithm will minimise
+    population_matrix = numpy.fromiter(
+        iter(-x for individual in population
+             for x in individual.fitness.wvalues),
+        dtype=numpy.float)
+    pop_len = len(population)
+    feat_len = len(population[0].fitness.wvalues)
+    population_matrix = population_matrix.reshape((pop_len, feat_len))
 
-    # Population size is the number of rows in the population_matrix
-    pop_size = len(population_matrix)
-
-    components_matrix = numpy.zeros((pop_size, pop_size))
-
-    # pylint: disable=F0401, E0611
-    import eps
-    # pylint: enable=F0401, E0611
-
-    # Calculator the indicator value for every element in the matrix
-    # The code inside this for loop is (has to be) heavily optimised for speed
-    for i in xrange(0, pop_size):
-        ind1 = population_matrix[i]
-        for j in itertools.chain(xrange(0, i), xrange(i + 1, pop_size)):
-            ind2 = population_matrix[j]
-            components_matrix[i, j] = eps.indicator(ind1,
-                                                    ind2,
-                                                    min_box_bounds,
-                                                    max_box_bounds)
+    # Calculate minimal square bounding box of the objectives
+    box_ranges = (numpy.max(population_matrix, axis=0) -
+                  numpy.min(population_matrix, axis=0))
+
+    components_matrix = numpy.zeros((pop_len, pop_len))
+    for i in xrange(0, pop_len):
+        diff = population_matrix - population_matrix[i, :]
+        components_matrix[i, :] = numpy.max(
+            numpy.divide(
+                diff,
```

Comment:
@mgeplf, you removed the indicator function here. We need to be able to switch this out with e.g. the hypervolume indicator. Is there a way to keep this indicator function call ?